interfaces/builtin/uhid.go

```diff
@@ -0,0 +1,128 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"bytes"
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/snapcore/snapd/interfaces"
+)
+
+const uhidConnectedPlugAppArmor = `
+# Description: Allows accessing the UHID to create kernel
+#  hid devices from user-space.
+
+  # Requires CONFIG_UHID
+  /dev/uhid                       rw,
+`
+
+type UhidInterface struct{}
+
+func (iface *UhidInterface) Name() string {
+	return "uhid"
+}
+
+func (iface *UhidInterface) String() string {
+	return iface.Name()
+}
+
+const uhidDeviceNode string = "/dev/uhid"
+
+// Check the validity of the slot
+func (iface *UhidInterface) SanitizeSlot(slot *interfaces.Slot) error {
+	// First check the type
+	if iface.Name() != slot.Interface {
+		panic(fmt.Sprintf("slot is not of interface %q", iface))
+	}
+
+	// Only os can create slot of this type
+	if slot.Snap.Type != "os" {
+		return fmt.Errorf("%s slots only allowed on core snaps", iface.Name())
+	}
```

Comment:
Perhaps I'm missing something here but the /dev/uhid a path itself so isn't it good to check it anyway? I know that the /dev/uhid is fixed however just for the sanity. What do you think?